pallets/index/src/lib.rs

```diff
@@ -210,58 +214,29 @@ pub mod pallet {
 				.collect();
 			ActivedTasks::<T>::insert(&worker, &new_task_queue);
 			let deposit_info = DepositRecords::<T>::get(&task_id).unwrap();
+			ensure!(deposit_info.amount > fee, Error::<T>::FeeTooExpensive);
 
 			// Withdraw from module account
 			let module_account: T::AccountId = MODULE_ID.into_account_truncating();
 			Self::do_asset_transact(
 				deposit_info.asset.clone(),
-				module_account,
+				module_account.clone(),
 				worker,
-				deposit_info.amount,
+				deposit_info.amount - fee,
```

Comment:
Have you considered ED problem?

Reply:
Yes, ED will be handled (checked) on the executor side